### DISCUSSION TITLE
webIO document.transform(draco()) dont work well

### DISCUSSION THREAD

**Discussion by Philbert-LX** [donmccurdy/glTF-Transform#1121] [Q&A] [answered]
I am using gltf to simplify and compress my 3D model, but I find that in WEBIO, my use of draco() doesn't seem to work, and the exported model is the same file size as if I didn't apply draco(). But I used the Command-line API (gltf-transform draco input.glb output.glb --method edgebreaker) to use DERCO on the simplified exported model file. So how do you make webio's draco () useful so that it is exported using DROCO compression
this my code
`js`
`const io = new WebIO({ credentials: 'include' });
  let document;
  document = await io.read('/models/aa.glb');  // → Document
  await document.transform(
    simplify({ simplifier: MeshoptSimplifier, ratio: 0.75, error: 0.001, lockBorder: false }),
    weld(
      { tolerance: 0.0001, toleranceNormal: 0.2 }
    ),
    quantize(),
    draco(),
  );
 // also dont work
   await document.createExtension(KHRDracoMeshCompression)
   .setRequired(true)
   .setEncoderOptions({
        method: KHRDracoMeshCompression.EncoderMethod.EDGEBREAKER,
        encodeSpeed: 5,
        decodeSpeed: 5,
    });

  const glb = await io.writeBinary(document); // Document → Uint8Array
  console.log(glb);
  exportGLB(glb)
function exportGLB(glbBuffer) {
  const blob = new Blob([glbBuffer], { type: 'model/gltf-binary' });
  // save(blob, "output.glb")
  const link = document.createElement('a');
  link.style.display = 'none';
  document.body.appendChild(link);
  link.href = URL.createObjectURL(blob);
  link.download = "output.glb";
  link.click();
}`

**Comment by donmccurdy** [comment, marked as answer]
Do you see any warnings in the JS console when running your code? I would expect there's something there about missing extensions in I/O. The Draco library needs to be installed separately - see:

- https://gltf-transform.dev/modules/extensions/classes/KHRDracoMeshCompression
- https://stackoverflow.com/questions/66975407/convert-glb-to-draco-glb-front-end/66978236#66978236

**Reply by Philbert-LX** [reply]
thanks very much
